### DISCUSSION TITLE
About Kalman filter

### DISCUSSION THREAD

**Discussion by Toach-2021** [talmolab/sleap#1581] [Help!]
Hello,

I have a very naive question, but how can I use Kalman filter for running inference with GUI?

When I run, I always get the error "Kalman filter requires target instance count".
I already prepared some predicted frames in the video.
I would appreciate any advice.

Thank you in advance.

**Comment by roomrys** [comment]
Hi @Toach-2021,

I wouldn't recommend using the Kalman Filters as they historically have not improved tracking, but if you would like to give them a try - then, I think the error message is telling you to set a Target Instance Count which would be the first input parameter under tracking  named "Cull to Target Instance Count".

Thanks,
Liezl

**Reply by Toach-2021** [reply]
Dear Liezl,

Thank you very much for your comment!
I would like to try the Kalman filter to smooth out the tracking because inevitably some individuals would not be detected or their locations would be blurred...

About Target Instance Count, I'm sorry but I could not find that option...
Maybe I missed but I would appreciate if you could tell me how to set it.

![スクリーンショット 2023-11-02 141809](https://github.com/talmolab/sleap/assets/148746416/0aeffe22-7db4-40c6-ae71-3707f1801ac2)

Best,

**Reply by roomrys** [reply]
Hi @Toach-2021,

Apologies, I was writing from memory, but we recently made some changes to these parameters. There used to be a "Cull to Target Instance Count" parameter and "Target Number of Instances Per Frame" parameter that has now been _replaced_ (in desired functionality) with "Max number of tracks" in #1447 - basically removing instances over a set limit earlier on in the tracking.

Although it looks like the variable names for these three parameters are different and although they have been removed from the GUI, the variables have not been refactored to reflect the preference for "Max number of Tracks".
https://github.com/talmolab/sleap/blob/0b2fdd9282dee40cd275bb040f91e575754bb048/sleap/config/pipeline_form.yaml#L379-L395

So, this is a GUI bug. I've created an issue to track this in #1583. However! Since there hasn't been changes to the variable names, using [`sleap-track`](https://sleap.ai/guides/cli.html?highlight=cli#sleap-track) through the CLI will work as intended. I could help you string together the command (given a screenshot of your GUI settings) if you need help. Alternatively, the GUI will output the `sleap-track` command that it is running (after you hit Run) in the terminal and you can copy/paste that command to tack on a `--tracking.target_instance_count <the_max_number_of_instances_per_frame>` and rerun it in the terminal (you will need to open a new terminal, activate the `sleap` environment, and then run the command via the terminal since the GUI will be using the currently open terminal). 

Sorry about that,
Liezl
